core/src/main/java/de/bwaldvogel/mongo/MongoBackend.java

```diff
@@ -40,12 +41,14 @@
 
     void close();
 
-    Clock getClock();
+    MongoBackendClock getClock();
 
-    void setClock(Clock clock);
+    void setClock(MongoBackendClock clock);
```

Comment:
perhaps we should not change the public API. should we overload the `setClock` so it can still accept Clock too? it can construct the `MongoBackendClock` internally.

Reply:
Yes, I think it is a better approach. Let's not expose clients to too many internal types